code/modules/client/preferences.dm

```diff
@@ -117,984 +117,983 @@ var/global/list/special_roles = list( //keep synced with the defines BE_* in set
 
-					if("next_facehair_style")
-						if (gender == MALE)
-							facial_hair_style = next_list_item(facial_hair_style, facial_hair_styles_male_list)
-						else
-							facial_hair_style = next_list_item(facial_hair_style, facial_hair_styles_female_list)
+				if("deity_name")
+					var/new_deity_name = reject_bad_name( input(user, "Choose your character's deity:", "Character Preference")  as text|null )
+					if(new_deity_name)
+						custom_names["deity"] = new_deity_name
+					else
+						user << "<font color='red'>Invalid name. Your name should be at least 2 and at most [MAX_NAME_LEN] characters long. It may only contain the characters A-Z, a-z, -, ' and .</font>"
 
-					if("previous_facehair_style")
-						if (gender == MALE)
-							facial_hair_style = previous_list_item(facial_hair_style, facial_hair_styles_male_list)
-						else
-							facial_hair_style = previous_list_item(facial_hair_style, facial_hair_styles_female_list)
 
-					if("underwear")
-						var/new_underwear
-						if(gender == MALE)
-							new_underwear = input(user, "Choose your character's underwear:", "Character Preference")  as null|anything in underwear_m
-						else
-							new_underwear = input(user, "Choose your character's underwear:", "Character Preference")  as null|anything in underwear_f
-						if(new_underwear)
-							underwear = new_underwear
-
-					if("undershirt")
-						var/new_undershirt
-						if(gender == MALE)
-							new_undershirt = input(user, "Choose your character's undershirt:", "Character Preference") as null|anything in undershirt_m
-						else
-							new_undershirt = input(user, "Choose your character's undershirt:", "Character Preference") as null|anything in undershirt_f
-						if(new_undershirt)
-							undershirt = new_undershirt
-
-					if("socks")
-						var/new_socks
-						if(gender == MALE)
-							new_socks = input(user, "Choose your character's socks:", "Character Preference") as null|anything in socks_m
-						else
-							new_socks = input(user, "Choose your character's socks:", "Character Preference") as null|anything in socks_f
-						if(new_socks)
-							socks = new_socks
-
-					if("eyes")
-						var/new_eyes = input(user, "Choose your character's eye colour:", "Character Preference") as color|null
-						if(new_eyes)
-							eye_color = sanitize_hexcolor(new_eyes)
-
-					if("species")
-
-						var/result = input(user, "Select a species", "Species Selection") as null|anything in roundstart_species
-
-						if(result)
-							var/newtype = roundstart_species[result]
-							pref_species = new newtype()
-							//Now that we changed our species, we must verify that the mutant colour is still allowed.
-							var/temp_hsv = RGBtoHSV(features["mcolor"])
-							if(features["mcolor"] == "#000" || (!(MUTCOLORS_PARTSONLY in pref_species.specflags) && ReadHSV(temp_hsv)[3] < ReadHSV("#7F7F7F")[3]))
-								features["mcolor"] = pref_specie
```

Comment:
The only real change done in this file. I removed this line. blood_type is now only in dna and updated here in the hardset_dna() on line 1096.